src/runTests.ts

```diff
@@ -1,64 +1,97 @@
-import * as path from 'path';
 import { blue, green, red, underline } from 'chalk';
+import * as path from 'path';
 import dirname, { projectName } from './dirname';
-import { TestArgs } from './main';
-import remapCoverage from './remapCoverage';
 
 const cs: any = require('cross-spawn');
 const pkgDir: any = require('pkg-dir');
 const packagePath = pkgDir.sync(dirname);
-/* Custom reporter used for reporting */
-const internReporter = `${path.relative(process.cwd(), packagePath)}/reporters/Reporter`;
 
 let logger = console.log;
 
-export function parseArguments({ all, config, functional, internConfig, reporters, secret, testingKey, unit, userName }: TestArgs) {
-	const configArg = config ? `-${config}` : '-local';
+export interface TestOptions {
+	nodeUnit?: boolean;
+	remoteUnit?: boolean;
+	remoteFunctional?: boolean;
+	childConfig?: string;
+	internConfig?: string;
+	reporters?: string;
+	userName?: string;
+	secret?: string;
+	testingKey?: string;
+	verbose?: boolean;
+	coverage?: boolean;
+	filter?: string;
+}
+
+export function parseArguments(testArgs: TestOptions) {
+	const {
+		nodeUnit,
+		remoteUnit,
+		remoteFunctional,
+		childConfig,
+		internConfig,
+		reporters,
+		secret,
+		testingKey,
+		userName,
+		filter
+	} = testArgs;
+
+	const configArg = childConfig ? `@${childConfig}` : '';
 	const args = [
 		internConfig
 			? `config=${path.relative(process.cwd(), internConfig)}`
-			: `config=${path.relative(process.cwd(), path.join(packagePath, 'intern', 'intern' + configArg))}`
+			: `config=${path.relative(process.cwd(), path.join(packagePath, 'intern', 'intern.json' + configArg))}`
 	];
 
-	if (!all && unit) {
+	// by default, in the intern config, all tests are run. we need to
+	// disable tests that we dont want to run
+	if (!nodeUnit) {
+		args.push('node={}');
+	}
+
+	if (!remoteUnit && !remoteFunctional) {
+		args.push('environments=');
+	}
+	else if (!remoteFunctional) {
 		args.push('functionalSuites=');
 	}
-	else if (functional) {
-		args.push('suites=');
+	else if (!remoteUnit) {
+		args.push('browser={}');
+	}
+
+	if (filter) {
+		args.push('grep=' + filter);
 	}
 
-	args.push(...(reporters ? reporters.split(',').map((reporter) => `reporters=${reporter}`) : [ `reporters=${internReporter}` ]));
+	args.push(...(reporters ? reporters.split(',').map((reporter) => `reporters=${reporter}`) : []));
 
-	if (config === 'testingbot' && testingKey && secret) {
-		args.push(`tunnelOptions={ "verbose": "true", "apiKey": "${testingKey}", "apiSecret": "${secret}" }`);
-		args.push(`webdriver={ "host": "http://hub.testingbot.com/wd/hub", "username": "${testingKey}", "accessKey": "${secret}" }`);
+	if (childConfig === 'testingbot' && testingKey && secret) {
+		args.push(`tunnelOptions={ "verbose": "true", "hostname": "hub.testingbot.com", "apiKey": "${testingKey}", "apiSecret": "${secret}" }`);
```

Comment:
The TestingBot tunnel uses the same auth credentials as the other tunnels (username and accessKey).